### ISSUE TITLE
hw05 ready for grading

### ISSUE THREAD

**Issue by abishekarun** [abishekarun/STAT545-hw-rajendran-arun#5]
[Rendered html file](http://htmlpreview.github.io/?https://github.com/abishekarun/STAT545-hw-rajendran-arun/blob/a4d2dbc7a63bfb086f6d41375609067234afec11/hw05/hw05_gapminder.html)

[hw05 folder](https://github.com/abishekarun/STAT545-hw-rajendran-arun/tree/a4d2dbc7a63bfb086f6d41375609067234afec11/hw05)

**Comment by santiagodr** [comment]
Hi @abishekarun Here are my comments:

- Your repo README files is well organized and linked to specific files and resources used in this homework. You described well what you did, issues you found, and specific additional resources.
- The factor management section is well explained with examples, comments and additional resources such as the `gdata` package. 
- I really like the examples and graphs you provided to explain the effect of reordering levels using `fct_reorder` and `fct_reorder2`, since I didn't explored this last one, and the use of `arrange( )`. A minor recommendation here is that in your *Ordered Plot of Mean population*, maybe it is better to keep ascending order, instead of `.desc = TRUE`, because the countries on top of the Y-axis are actually the ones with smaller populations, it doesn't really matter, but I guess it makes more sense to see an ascending "pattern" here, since the X-axis is ascending too.
- File I/O and Visualization design sections are clear and well explained, including several options for exporting/importing and a clear improvement of a previous graph. 
- I also learned from your assignment  about other function in the forcats package such as `fct_rev`, `fct_expand` and `fct_recode`.
- Overall excellent job! It is a nice and comprehensive assignment!

**Comment by ksedivyhaley** [comment]
**Factor management (drop & reorder):** Yes
**File I/O (data):** Yes
**Visualization design:** Yes (see comments)
**File I/O (write figure):** Yes
**Organized GitHub:** OK (see comments)
**Bonus (more forcats, eg relevel):** Yes
**Reflection:** Yes

**Comments:**

- For your repo, add “slugs” to the files, eg hw05_FactorsFigures, so you know what is in each assignment (especially since the table of contents in your README doesn't include this information). 
- Good job using folders for your extra homework files.
- Good discussion of dropping factor levels, and the effect of factor reordering on a figure - thorough exploration of forcats.
- Note that `fct_reorder(gapminder$country, gapminder$pop, mean)` does not actually perform the reodering on the gapminder$country data – it makes a new factor.
- What does setting a factor value to NULL do to the occurrences of that value within the dataset?
- Once you've filtered the data to only 2007 data, the `fun = mean` argument in `fct_reorder(gap_asia$country, gap_asia$pop, mean)` is non-functional since you have only one data point for each country – thus it's not really meaningful to order by the country's mean GDP value. 
- Thorough exploration of data & figure I/O
- For visualization design, good job setting the breaks and theme, and rotating the x-axis labels. expaand_limits is also a good touch. Good work executing the colour change, though in this case I don't think it's very informative; in fact, it mostly makes it hard to read the plots for lower live expectancy values (Africa in particular). You could also set the title text for your x- and y-axis titles, and your legend.

Your mark will be distributed later. If you would like more feedback, please feel free to message me on slack.

**Comment by ssheikho** [comment]
Hi @abishekarun ! I understand you've closed this issue, but I finally got a chance to review your work. I apologize for being so late!

Strength:
+ Most tasks in the assignment were done very well. In particular, you thoroughly explored factor reordering and leveling data, and file/figure I/O. 
+ I really like your graphs, specially the first two!
+ I really like how you provide links to the resources that helped you through the homework

Things to work on:
- I found it hard to navigate through your homework. Maybe add a table of content to your homework, and divide up the homework into (numbered) sections that your toc links to. 
- In your README file add links to the homework, and maybe each section in the homework (like a reference table).
- in your "Plot of lifeExp vs year for Continent", maybe choose few top countries from each continent, and add a legend for each graph to specify the countries? Currently, the reader can't tell which line refers to which country!

Overall, really good job on the homework :)
